crates/nu-cli/src/completions/dotnu_completions.rs

```diff
@@ -140,70 +93,50 @@ impl Completer for DotNuCompletion {
         }
 
+        // Fetch the files
+        completions.extend(complete_item(
+            false,
+            span,
+            prefix.as_ref(),
+            &search_dirs
+                .iter()
+                .filter_map(|d| d.to_str())
+                .collect::<Vec<_>>(),
+            options,
+            working_set.permanent_state,
+            stack,
+        ));
+
+        let into_suggestion = |x: &FileSuggestion| SemanticSuggestion {
+            suggestion: Suggestion {
+                value: x.path.to_string(),
+                style: x.style,
+                span: reedline::Span {
+                    start: x.span.start - offset,
+                    end: x.span.end - offset,
+                },
+                append_whitespace: !x.is_dir,
+                ..Suggestion::default()
+            },
+            kind: Some(SuggestionKind::Module),
+        };
+
+        // Put files atop
         completions
-            .into_iter()
-            // Different base dir, so we list the .nu files or folders
+            .iter()
+            // filtering the files that ends with .nu
             .filter(|it| {
                 // for paths with spaces in them
                 let path = it.path.trim_end_matches('`');
-                path.ends_with(".nu") || it.is_dir
-            })
-            .map(|x| {
-                let append_whitespace = !x.is_dir && (!start_with_backquote || end_with_backquote);
-                // Re-calculate the span to replace
-                let mut span_offset = 0;
-                let mut value = x.path.to_string();
-                // Complete only the last path component
-                if base_dir == MAIN_SEPARATOR_STR {
-                    span_offset = base_dir.len()
-                } else if base_dir != "." {
-                    span_offset = base_dir.len() + 1
-                }
-                // Retain only one '`'
-                if start_with_backquote {
-                    value = value.trim_start_matches('`').to_string();
-                    span_offset += 1;
-                }
-                // Add the backquote back
-                if end_with_backquote && !value.ends_with('`') {
-                    value.push('`');
-                }
-                let end = x.span.end - offset;
-                let start = std::cmp::min(end, x.span.start - offset + span_offset);
-                SemanticSuggestion {
-                    suggestion: Suggestion {
-                        value,
-                        style: x.style,
-                        span: reedline::Span { start, end },
-                        append_whitespace,
-                        ..Suggestion::default()
-                    },
-                    kind: Some(SuggestionKind::Module),
-                }
+                path.ends_with(".nu")
             })
+            // or directories
+            .chain(completions.iter().filter(|it| it.is_dir))
+            .map(into_suggestion)
```

Comment:
I'm not sure pulling `into_suggestion` out to a closure is necessary, but it's fine either way